README.md

```diff
@@ -33,40 +38,81 @@ node example/klaytn_tx_ethers.js
 
 ```mermaid
 classDiagram
-  TypedFields <|-- TypedTx
-  TypedFields <|-- TypedAccountKey
-  class TypedFields {
-    static type: number
-    static fieldTypes: string -> FieldType
+  FieldType ..|> FieldTypeBytes
+  FieldType ..|> FieldTypeSignatureTuples
+  FieldType ..|> FieldTypeAccountKey
+  FieldType ..|> etc
+  class FieldType {
+    <<interface>> 
+    canonicalize(any): any
+    emptyValue(): any
+  }
+  class FieldTypeBytes {
+    canonicalize(any): string
+    emptyValue(): string
+  }
+  class FieldTypeSignatureTuples {
+    canonicalize( SignatureLike[]): SignatureTuple[]
+    emptyValue(): SignatureTuple[]
+  }
+  class FieldTypeAccountKey {
+    canonicalize(TypedAccountKey | string | any): string
+    emptyValue(): string
+  }
+```
+
+```mermaid  
+classDiagram
+  FieldSet <|-- KlaytnTx
+  KlaytnTx <|-- TxTypeValueTransfer
+  KlaytnTx <|-- TxTypeFeeDelegatedValueTransfer
+  KlaytnTx <|-- other TxTypes
+  FieldSet <|-- AccountKey
+  AccountKey <|-- AccountKeyLegacy
+  AccountKey <|-- AccountKeyPublic
+  AccountKey <|-- other AccountKey
+  class FieldSet {
+    type: number
+    typeName: string
+    fieldTypes: string -> FieldType
     setFields(any)
-    getField(string): any
+    setFieldsFromArray( string[], any[] )
+    getField( string ): any
+    getFields( string[] ): any[]
     toObject(): any
   }
-  class TypedTx {
+  class KlaytnTx {
     sigRLP(): string
     sigFeePayerRLP(): string
     senderTxHashRLP(): string
-    txRLP(): string
+    txHashRLP(): string
     addSenderSig(sig)
     addFeePayerSig(sig)
+    setFieldsFromRLP(string): void
   }
-  class TypedAccountKey {
+  class AccountKey {
     toRLP(): string
   }
+```
 
-  TypedFieldsFactory <|.. TypedTxFactory
-  TypedFieldsFactory <|.. TypedAccountKeyFactory
-  class TypedFieldsFactory {
+```mermaid  
+classDiagram
+  FieldSetFactory <|.. KlaytnTxFactory
+  FieldSetFactory <|.. AccountKeyFactory
+  class FieldSetFactory {
+    private registry: [number] -> FieldSet
+    private requiredFields: string[]
     add(typeof T)
+    has(type?): boolean
+    lookup(type?): typeof T
     fromObject(any): T
   }
-  class TypedTxFactory {
-    add(typeof TypedTx)
-    fromObject(any): TypedTx
+  class KlaytnTxFactory {
+    fromRLP(string): KlaytnTx
   }
-  class TypedAccountKeyFactory {
-    add(typeof TypedAccountKey)
-    fromObject(any): TypedAccountKey
+  class AccountKeyFactory {
+    canonicalize(AccountKey | string | any): string 
+    emptyValue(): string
```

Comment:
AccountKeyFactory does not have additional methods compared to FieldSetFactory.

Reply:
After implementing AccountKey, if there is still no function to add, we will replace it with FieldSetFactory.

Reply:
I see.
But this diagram seems wrong as-is. canonicalize and emptyValue are methods of FieldTypeAccountKey, not AccountKeyFactory.

Reply:
Oops
I was confused with FieldTypeAccountKey.
thank you.

https://github.com/blukat29/klaytn-sdk-design/pull/2/commits/2c94b40b11de7eff3e3680f08789303ea0f0fbe9